pennylane_lightning/lightning_qubit/_adjoint_jacobian.py

```diff
@@ -0,0 +1,362 @@
+        return tuple(tuple(np.array(j_) for j_ in j) for j in jac)
+
+    def calculate_jacobian(self, tape: QuantumTape):
+        """Computes the Jacobian with the adjoint method.
+
+        .. code-block:: python
+
+            statevector = LightningStateVector(num_wires=num_wires)
+            statevector = statevector.get_final_state(tape)
+            jacobian = LightningAdjointJacobian(statevector).calculate_jacobian(tape)
+
+        Args:
+            tape (QuantumTape): Operations and measurements that represent instructions for execution on Lightning.
+
+        Returns:
+            The adjoint Jacobian of a tape.
+        """
+
+        if tape.shots:
+            warn(
+                "Requested adjoint differentiation to be computed with finite shots. "
+                "The derivative is always exact when using the adjoint "
+                "differentiation method.",
+                UserWarning,
+            )
+
+        tape_return_type = self._check_adjdiff_supported_measurements(tape.measurements)
+
+        if not tape_return_type:  # the tape does not have measurements
+            return np.array([], dtype=self._dtype)
+
+        if tape_return_type is State:
+            raise QuantumFunctionError("This method does not support statevector return type. ")
+
+        self._check_adjdiff_supported_operations(tape.operations)
+
+        processed_data = self._process_jacobian_tape(tape)
+
+        if not processed_data:  # training_params is empty
+            return np.array([], dtype=self._dtype)
+
+        trainable_params = processed_data["tp_shift"]
+
+        # If requested batching over observables, chunk into OMP_NUM_THREADS sized chunks.
+        # This will allow use of Lightning with adjoint for large-qubit numbers AND large
+        # numbers of observables, enabling choice between compute time and memory use.
+        requested_threads = int(getenv("OMP_NUM_THREADS", "1"))
+
+        if self._batch_obs and requested_threads > 1:
+            obs_partitions = _chunk_iterable(processed_data["obs_serialized"], requested_threads)
+            jac = []
+            for obs_chunk in obs_partitions:
+                jac_local = self._jacobian_lightning(
+                    processed_data["state_vector"],
+                    obs_chunk,
+                    processed_data["ops_serialized"],
+                    trainable_params,
+                )
+                jac.extend(jac_local)
+        else:
+            jac = self._jacobian_lightning(
+                processed_data["state_vector"],
+                processed_data["obs_serialized"],
+                processed_data["ops_serialized"],
+                trainable_params,
+            )
+        jac = np.array(jac)
+        jac = jac.reshape(-1, len(trainable_params))
+        jac_r = np.zeros((jac.shape[0], processed_data["all_params"]))
+        jac_r[:, processed_data["record_tp_rows"]] = jac
+        if hasattr(qml, "active_return"):  # pragma: no cover
```

Comment:
I don't think we need to worry about the old return types specification anymore.  I don't think it will make sense to be using this version of lightning with pennylane from over a year ago.